Add option for LoA header level

psomhorst opened the pull request:
Adds a metadata option `loa_header_level` and shortcode option `header_level`. Defaults to 1 (current default).

This is an easy solve for an issue where the default LoA header level (1) does not match the header level Quarto teaches (2). Also, makes it easier to add the LoA as a subsection. 

Fixes #49.

Commit:
Add option for LoA header level

rchaput commented:
This is a great idea! That simplifies changing the LoA (the example given in the documentation is quite verbose for such a simple change) without changing the default...

I'll need to write the documentation, tests, and I'd also like to change a bit the way you raise errors to be coherent with the other errors in **acronyms** (although I should change them as well at some point). I'll try to do it when I have time, probably this weekend.

Commit:
Update error detection in option loa_header_level

Commit:
Add test for option loa_header_level

Commit:
Update docs to add option loa_header_level

Commit:
Fix header_level option when generating the LoA

- Parse the `header_level` keyworded-argument from the shortcodes and
  pass it to the `generateLoA` function. (The argument was in the
  signature but never given). Unfortunately, for the filter
  `\printacronyms`, we still do not parse arguments as we do for
  acronyms.

- Changed the way we check the value and raise an error if it does not
  correspond to a number, for consistency with other errors in acronyms.

rchaput commented:
I've fixed the way you set `header_level` when generating the List of Acronyms (allowing it to be set directly from the shortcode, in addition to the document metadata); added tests and documentation.

Merging this PR
